python/cudaq/kernel/ast_bridge.py

```diff
@@ -2720,6 +2720,16 @@ def visit_Compare(self, node):
             return
 
         if isinstance(op, ast.NotEq):
+            if F64Type.isinstance(left.type) and IntegerType.isinstance(
+                    comparator.type):
+                left = arith.FPToSIOp(comparator.type, left).result
```

Comment:
I'm not following why the LHS, if it's a floating point value, is integer truncated here.

Reply:
This is existing code from the `isinstance(op, ast.Eq)` below, which, in principle, should also be handled in the `ast.NotEq` case here.

I don't know the context of it so didn't change it or migrate it to `cc.CastOp` yet. 
@khalatepradnya Do you know the intent of this float -> int conversion in equal check?

Reply:
nope, sorry. Tagging @amccaskey for details.
I see that the `comparator` is of integer type and that the integer comparator is being used (`arith.cmpi`) on line#2740

Reply:
I'm wondering if this isn't a bug. Consider this python code
```python
>>> a=1
>>> b=1.2
>>> c= b==a
>>> print(c)
False
```